### ISSUE TITLE
quasar init is broken (Webpack4 / 0.16.0-beta.2)

### ISSUE THREAD

**Issue by nothingismagick** [quasarframework/quasar-cli#118]
tl;dr

quasar info:

```
NodeJs                          10.0.0

Global packages                 
  NPM                           5.6.0
  yarn                          1.6.0
  quasar-cli                    0.16.0-beta.2
  vue-cli                       3.0.0-beta.1
  cordova                       8.0.0

Important local packages        
  quasar-cli                    0.16.0-beta.2   (Quasar Framework CLI)
  @babel/core                   Not available
  workbox-webpack-plugin        Not available
```

babel core and workbox cannot be found on a vanilla 0.16.0-beta.2

---------
The first seems to be a warning from yarn.
(node:3862) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.

This eslint-loader issue has been around for a while now... Also not really anything we can fix. There are other ([perhaps related issues at the repo](https://github.com/webpack-contrib/eslint-loader/issues/227)).
```
[4/5] 🔗  Linking dependencies...
warning " > eslint-loader@2.0.0" has unmet peer dependency "webpack@>=2.0.0 <5.0.0".
[5/5] 📃  Building fresh packages...
```

This is my quasar info immediately afterwards (I am building with v0.16.0.beta.2)

```bash
quasar info

Operating System                Darwin(16.7.0) - darwin/x64
NodeJs                          10.0.0

Global packages                 
  NPM                           5.6.0
  yarn                          1.6.0
  quasar-cli                    0.15.14
  vue-cli                       3.0.0-beta.1
  cordova                       8.0.0

Important local packages        
  quasar-cli                    0.15.14 (Quasar Framework CLI)
  quasar-framework              0.15.10 (Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras                 1.0.2   (Quasar Framework fonts, icons and i18n.)
  vue                           2.5.16  (Reactive, component-oriented view layer for modern web interfaces.)
  vue-router                    3.0.1   (Official router for Vue.js 2)
  vuex                          3.0.1   (state management for Vue.js)
  electron                      Not available
  babel-core                    6.26.0  (Babel compiler core.)
  webpack                       3.11.0  (Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server            2.11.1  (Serves a webpack app. Updates the browser on changes.)
```

Then I `yarn link quasar-cli` (to the local version of quasar-cli I am building for the test mode)
```bash
quasar info

Operating System                Darwin(16.7.0) - darwin/x64
NodeJs                          10.0.0

Global packages                 
  NPM                           5.6.0
  yarn                          1.6.0
  quasar-cli                    0.16.0-beta.2
  vue-cli                       3.0.0-beta.1
  cordova                       8.0.0

Important local packages        
  quasar-cli                    0.16.0-beta.2   (Quasar Framework CLI)
  quasar-framework              0.15.10 (Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras                 1.0.2   (Quasar Framework fonts, icons and i18n.)
  vue                           2.5.16  (Reactive, component-oriented view layer for modern web interfaces.)
  vue-router                    3.0.1   (Official router for Vue.js 2)
  vuex                          3.0.1   (state management for Vue.js)
  electron                      Not available
  @babel/core                   Not available
  webpack                       3.11.0  (Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server            2.11.1  (Serves a webpack app. Updates the browser on changes.)
  workbox-webpack-plugin        Not available
```
Notable is that both `@babel/core` and `workbox-webpack-plugin` are no longer available.

But they are in the package.json:
```json
    "@babel/core": "^7.0.0-beta.46",
    "workbox-webpack-plugin": "^3.2.0"
```
and yarn should find them:   e.g. @babel/core is "version": "7.0.0-beta.46",
and yarn install says: `success Already up-to-date.`

I should know better by now, but then I `quasar dev` and:

```
$ quasar dev
 app:dev Running: Mode [ SPA ] with [ MAT ] theme +0ms

 app:quasar-conf Reading quasar.conf.js +2s
 app:dev Checking listening address availability (0.0.0.0:8080)... +4ms
 app:quasar-conf Generating Webpack config +7ms

 WARNING
 Your newer Quasar CLI requires a change to .babelrc file.
 Doing it automatically. Please review the changes.

 app:quasar-conf Extending Webpack config +35ms
 app:generator Generating Webpack entry point +5ms
 app:dev-server Booting up... +4ms

Build completed in 1.521s



 ERROR  Failed to compile with 1 errors                                                                                       2:06:31 AM

 error  in ./.quasar/entry.js

Module build failed:   ReferenceError: Unknown plugin "@babel/transform-runtime" specified in "/Users/denjell/WebstormProjects/quasarTest2/test3/beta/.babelr  c" at 0, attempted to resolve relative to "/Users/denjell/WebstormProjects/quasarTest2/test3/beta"
  
  - option-manager.js:180 
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:180:17
  
  - Array.map
  
  - option-manager.js:158 Function.normalisePlugins
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:158:20
  
  - option-manager.js:234 OptionManager.mergeOptions
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:234:36
  
  - option-manager.js:368 OptionManager.init
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:368:12
  
  - index.js:212 File.initOptions
    [beta]/[babel-core]/lib/transformation/file/index.js:212:65
  
  - index.js:135 new File
    [beta]/[babel-core]/lib/transformation/file/index.js:135:24
  
  - pipeline.js:46 Pipeline.transform
    [beta]/[babel-core]/lib/transformation/pipeline.js:46:16
  
  - index.js:50 transpile
    [beta]/[babel-loader]/lib/index.js:50:20
  
  - index.js:173 Object.module.exports
    [beta]/[babel-loader]/lib/index.js:173:20
  

 @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 webpack/hot/dev-server ./.quasar/entry.js

✖ ｢wdm｣: Hash: 472179a07df436fda117
Version: webpack 4.6.0
Time: 1529ms
Built at: 05/05/2018 2:06:31 AM
 Asset     Size  Chunks  Chunk Names
app.js  875 KiB     app  app
Entrypoint app = app.js
[../../../quasar-cli-OFFICIAL/node_modules/node-libs-browser/node_modules/punycode/punycode.js] /Users/denjell/WebstormProjects/quasar-cli-OFFICIAL/node_modules/node-libs-browser/node_modules/punycode/punycode.js 14.3 KiB {app} [built]
[../../../quasar-cli-OFFICIAL/node_modules/url/url.js] /Users/denjell/WebstormProjects/quasar-cli-OFFICIAL/node_modules/url/url.js 22.8 KiB {app} [built]
[../../../quasar-cli-OFFICIAL/node_modules/webpack-dev-server/client/index.js?http://0.0.0.0:8080] (webpack)-dev-server/client?http://0.0.0.0:8080 7.75 KiB {app} [built]
[../../../quasar-cli-OFFICIAL/node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.58 KiB {app} [built]
[../../../quasar-cli-OFFICIAL/node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.05 KiB {app} [built]
[./.quasar/entry.js] 3.28 KiB {app} [built] [failed] [1 error]
[./node_modules/ansi-html/index.js] 4.16 KiB {app} [built]
[./node_modules/loglevel/lib/loglevel.js] 7.68 KiB {app} [built]
[./node_modules/strip-ansi/index.js] 161 bytes {app} [built]
[./node_modules/webpack/hot sync ^\.\/log$] ./node_modules/webpack/hot sync nonrecursive ^\.\/log$ 170 bytes {app} [built]
[./node_modules/webpack/hot/dev-server.js] 1.58 KiB {app} [built]
[./node_modules/webpack/hot/emitter.js] 77 bytes {app} [built]
[0] multi (webpack)-dev-server/client?http://0.0.0.0:8080 webpack/hot/dev-server ./.quasar/entry.js 52 bytes {app} [built]
[./node_modules/webpack/hot/log-apply-result.js] 1.28 KiB {app} [built]
[./node_modules/webpack/hot/log.js] 1.02 KiB {app} [built]
    + 13 hidden modules

ERROR in ./.quasar/entry.js
Module build failed:   ReferenceError: Unknown plugin "@babel/transform-runtime" specified in "/Users/denjell/WebstormProjects/quasarTest2/test3/beta/.babelr  c" at 0, attempted to resolve relative to "/Users/denjell/WebstormProjects/quasarTest2/test3/beta"
  
  - option-manager.js:180 
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:180:17
  
  - Array.map
  
  - option-manager.js:158 Function.normalisePlugins
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:158:20
  
  - option-manager.js:234 OptionManager.mergeOptions
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:234:36
  
  - option-manager.js:368 OptionManager.init
    [beta]/[babel-core]/lib/transformation/file/options/option-manager.js:368:12
  
  - index.js:212 File.initOptions
    [beta]/[babel-core]/lib/transformation/file/index.js:212:65
  
  - index.js:135 new File
    [beta]/[babel-core]/lib/transformation/file/index.js:135:24
  
  - pipeline.js:46 Pipeline.transform
    [beta]/[babel-core]/lib/transformation/pipeline.js:46:16
  
  - index.js:50 transpile
    [beta]/[babel-loader]/lib/index.js:50:20
  
  - index.js:173 Object.module.exports
    [beta]/[babel-loader]/lib/index.js:173:20
  
 @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 webpack/hot/dev-server ./.quasar/entry.js
Child html-webpack-plugin for "index.html":
         Asset      Size  Chunks  Chunk Names
    index.html  1.39 MiB       0  
    Entrypoint undefined = index.html
    [../../../quasar-cli-OFFICIAL/node_modules/html-webpack-plugin/lib/loader.js!./src/index.template.html] /Users/denjell/WebstormProjects/quasar-cli-OFFICIAL/node_modules/html-webpack-plugin/lib/loader.js!./src/index.template.html 3.49 KiB {0} [built]
    [../../../quasar-cli-OFFICIAL/node_modules/lodash/lodash.js] /Users/denjell/WebstormProjects/quasar-cli-OFFICIAL/node_modules/lodash/lodash.js 527 KiB {0} [built]
    [../../../quasar-cli-OFFICIAL/node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 489 bytes {0} [built]
    [../../../quasar-cli-OFFICIAL/node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
```

**Comment by nothingismagick** [comment]
here is the fix: forget yarn link. that won't work. in the root directory of your git cloned quasar-cli repo:
```
$ ./bin/quasar init nolink
...
cd nolink
$ .././bin/quasar info
Operating System                Darwin(16.7.0) - darwin/x64
NodeJs                          10.0.0

Global packages                 
  NPM                           5.6.0
  yarn                          1.6.0
  quasar-cli                    Not available
  vue-cli                       3.0.0-beta.1
  cordova                       8.0.0

Important local packages        
  quasar-cli                    0.15.14 (Quasar Framework CLI)
  quasar-framework              0.15.10 (Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras                 1.0.2   (Quasar Framework fonts, icons and i18n.)
  vue                           2.5.16  (Reactive, component-oriented view layer for modern web interfaces.)
  vue-router                    3.0.1   (Official router for Vue.js 2)
  vuex                          3.0.1   (state management for Vue.js)
  electron                      Not available
  babel-core                    6.26.0  (Babel compiler core.)
  webpack                       3.11.0  (Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server            2.11.1  (Serves a webpack app. Updates the browser on changes.)

$ quasar dev
```
![screen shot 2018-05-05 at 02 30 29](https://user-images.githubusercontent.com/35242872/39657845-524f5036-500c-11e8-9589-a07ebd101d48.png)

**Comment by nothingismagick** [comment]
guess what I am now trying:
![screen shot 2018-05-05 at 02 34 35](https://user-images.githubusercontent.com/35242872/39657905-f69bb328-500c-11e8-96d5-59751090a9b1.png)

**Comment by nothingismagick** [comment]
But it still can't find @babel/core or workbox-webpack-plugin

I must be missing something. Like sleep
